master/buildbot/db/logs.py

```diff
@@ -329,6 +329,8 @@ def thdcompressLog(conn):
             q = sa.select([sa.func.sum(sa.func.length(tbl.c.content))])
             q = q.where(tbl.c.logid == logid)
             newsize = conn.execute(q).fetchone()[0]
+            if newsize is None:
+                newsize = totlength
```

Comment:
On my patch I just used 0

newsize = conn.execute(q).fetchone()[0] or 0

but it could just as easily be:

newsize = conn.execute(q).fetchone()[0] or totlength

I don't know which would be more correct.